### ISSUE TITLE
Assertion violation at src/smtsolvers/Theory.cc:202

### ISSUE THREAD

**Issue by rainoftime** [usi-verification-and-security/opensmt#78]
Hi, for the following formula,
~~~~
(set-logic QF_LIA)
(declare-fun v18 () Bool)
(declare-fun i2 () Int)
(declare-fun i3 () Int)
(push 1)
(assert (= 69 (+ 4 i3)))
(check-sat)
(assert v18)
(assert (= i2 66))
(check-sat)
(check-sat)
(check-sat)
(check-sat)
(push 1)
(assert (= 71 (+ 4 i3)))
(push 1)
(check-sat)
(check-sat)
(pop 1)
(check-sat)
(pop 1)
(check-sat)
(check-sat)
(pop 1)
(check-sat)
(check-sat)
(check-sat)
(check-sat)
(assert (= false (= 66 i3)))
(check-sat)
~~~~

opensmt (commit 04c7fa3) throws an assertion violation
~~~~
sat
sat
sat
sat
sat
unsat
unsat
unsat
sat
sat
sat
sat
sat
sat
opensmt: /home/rainoftime/Work/tofuzz/opensmt/src/smtsolvers/Theory.cc:202: TPropRes CoreSMTSolver::handleUnsat(): Assertion `max_decision_level <= decisionLevel( )' failed.
Aborted (core dumped)
~~~~

**Comment by blishko** [comment]
This one now crashes in a different place:
Assertion failed: (safeValue(l1) == l_Undef || safeValue(l2) == l_Undef), function handleSat, file /Users/mblicha/CLionProjects/opensmt2/src/smtsolvers/Theory.cc, line 102.

The current implementation of LIA has some invariants that might be broken in incremental solving.
Here the problem is that a SAT solver variable corresponding to an inequality from popped frame is kept in the SAT solver, but the clauses are already ignored and the theory solver actually ignores the inequality.

This is already unusual. The problem manifests when LIA solver wants to branch and one of the branch inequalities is actually the popped inequality. This inequality is not asserted in the theory solver, but its corresponding SAT solver variable is already assigned to true. This situation leads to the detected assertion violation.

**Comment by blishko** [comment]
Fixed by commit 68e3a29f in #111
